### ISSUE TITLE
Naming anonymous functions sometimes causes conflicts.

### ISSUE THREAD

**Issue by ProLoser** [babel/babel#1799]
So, I was coding a bit of an odd edge-case and ran into a problem where auto-naming anonymous functions was breaking code:

```js
// outer
function(widget) {
  doSomething({
    // inner
    widget: function() {
      // returns outer widget
      return widget;
    }
  });
}
```

This was causing an error because the outer `widget` was being renamed to `_widget` along with the return line because the `widget: function()` becomes `widget: function widget()`. Since this name is being autogenerated, the variables I explicitly declare should take precedence. In AngularJS, the name of a function's parameters are introspected to handle dependency injection, and the parameters are being renamed through babel. Normally, you run your angular code through a preparation filter (ng-min) that avoids this sort of issue, but the problem is those preparation filters don't understand ES6 so I have to compile in babel BEFORE escaping in my angular minification filter. 

## tl;dr:
Let explicit variable names take precedence over auto-generated anonymous function names.

**Comment by monsanto** [comment]
that's not possible. it HAS to be `function widget()` for the correct function `name` property to be set. If you don't care about the `name` property being correct you can blacklist the transformer `spec.functionName`
